rs_core/src/main/java/com/luxoft/jmswithspring/model/CountryCodeHandler.java

```diff
@@ -1,16 +1,21 @@
 package com.luxoft.jmswithspring.model;
 
+import com.luxoft.jmswithspring.model.exception.InvalidCountryCodeException;
+
 import javax.xml.bind.annotation.adapters.XmlAdapter;
 
 public class CountryCodeHandler extends XmlAdapter<Integer, CountryCode>  {
 
     @Override
-    public CountryCode unmarshal(Integer v) throws Exception {
-        return CountryCode.match(v);
+    public CountryCode unmarshal(Integer v) throws InvalidCountryCodeException {
+        CountryCode code = CountryCode.match(v);
+        if(code != null)
```

Comment:
Optional.of(code).orElseThrow(...)

Always use if{...} else {...} never not if ...; ...;